modules/gcloud/src/main/java/org/testcontainers/containers/SpannerEmulatorContainer.java

```diff
@@ -0,0 +1,28 @@
+package org.testcontainers.containers;
+
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+import org.testcontainers.utility.DockerImageName;
+
+/**
+ * A Spanner container. Default ports: 9010 for GRPC and 9020 for HTTP.
+ *
+ * @author Eddú Meléndez
+ */
+public class SpannerEmulatorContainer extends GenericContainer<SpannerEmulatorContainer> {
+
+    private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse("gcr.io/cloud-spanner-emulator/emulator");
+
+    private static final int GRPC_PORT = 9010;
+    private static final int HTTP_PORT = 9020;
```

Comment:
I wonder, maybe these should be public constants so that they can be used by tests? If we did that, we could do so for the other container classes in this PR.

WDYT @bsideup, @kiview?

Reply:
WDYT about not exposing the constants, but `getGrpcPort`/`getHttpPort` methods?

Reply:
looking at some of the other modules, `private static final` seems like the norm/convention atm. would it make sense to follow up w/ an issue for the broader change to expose these across modules?

Reply:
I've been persuaded by @bsideup - many of our other modules have some mechanism for getting the mapped ports, address or URL for the running container. It seems like we should do this here, instead of exposing constants, as it's fundamentally more useful for the user.

I'll take the action to do this, as we've asked @eddumelendez to do too much already. I'll raise a quick PR tonight, and will merge this PR now.